### ISSUE TITLE
Cannot specify container work dir when running `ansible-runner` via cmdline

### ISSUE THREAD

**Issue by vrubiolo** [ansible/ansible-runner#945]
Hi guys,

[Based on advice from @AlanCoding on [issues/758](https://github.com/ansible/ansible-runner/issues/758), I am opening a separate issue to track my issue. I also posted about the issue on https://groups.google.com/g/awx-project/c/Il-q0mfkoeA and will reference this ticket there.]

I am focusing on running `ansible-runner` locally in commandline mode on our existing repositories with process isolation done via docker (so as to try and match what AWX does).

This playbook repository has a particular structure which does not abide by the ansible-runner input directory hierarchy (https://ansible-runner.readthedocs.io/en/stable/intro/#inputdir). I am thus using the several ansible-runner commandline parameters to point at the various pieces that are required. 

I am stumbling upon an issue in that the playbook I specify cannot be found by the newly spawned container:
ERROR! the playbook: deploy-stack.yaml could not be found

Investigating in debug mode [log attached](https://github.com/ansible/ansible-runner/files/7747660/ansible-runner-log.txt), I see this is because the underlying docker commandline always has `--workdir /runner/project` specified whereas the playbook is actually located/mounted under `/runner/deploy-stack.yaml`

If I edit the docker commandline and run it manually to specify `--workdir /runner`, then `ansible-playbook` starts and find the playbook, as expected.

I have looked into the `ansible-runner` source (I am using v2.1.1) and it seems the logic which sets the working directory is at https://github.com/ansible/ansible-runner/blob/2.1.1/ansible_runner/config/_base.py#L451

It should be overridable via either `self.container_dir` or `self.host_cwd`. The thing is, I can't find where the commandline entry point (and not the Python one) is actually setting those values. It looks like this option is missing.

Could someone explain how to specify the container working directory when running `ansible-runner` from the commandline?

**Comment by kurokobo** [comment]
In my understanding, currently [there is no option to override `container_workdir` or `host_cwd`](https://github.com/ansible/ansible-runner/blob/d01304455c3c25669fba70d9983487ffdb85f1c5/ansible_runner/__main__.py#L869-L906).

As a workaround, this is hacky and dirty way, but you can make your current directory mounted as `/runner/project` by adding `--container-volume-mount .:/runner/project`.

```bash
# Only inventory and playbooks are here
$ ls -l 
total 8
-rw-rw-r--. 1 kuro kuro 193 Dec 14 19:05 inventory
-rw-rw-r--. 1 kuro kuro  99 Dec 21 22:13 site.yml

# But we can run playbook by one-liner with ansible-runner
$ ansible-runner run --process-isolation --process-isolation-executable docker --container-image registry.example.com/ansible/ee:2.10-custom --container-volume-mount .:/runner/project . -p site.yml
...
PLAY [localhost] ***************************************************************

TASK [ansible.builtin.ping] ****************************************************
ok: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

The `project` and `artifacts` directories will be created in the current directory by ansible-runner, but it works anyway.

```bash
$ ls -l
total 8
drwxrwxr-x. 4 kuro kuro  94 Dec 21 22:13 artifacts
-rw-rw-r--. 1 kuro kuro 193 Dec 14 19:05 inventory
drwxr-xr-x. 2 root root   6 Dec 21 22:12 project
-rw-rw-r--. 1 kuro kuro  99 Dec 21 22:13 site.yml
```

Tested on Docker 20.10.7 on CentOS 8.2, with ansible-runner 2.1.1. Not tested with Podman.

**Comment by Shrews** [comment]
It does appear that we do have this setting available via the API (`container_dir`), but we do not expose it for the CLI. This is probably simple to implement.

**Comment by vrubiolo** [comment]
@kurokobo : thanks for the workaround! I have tried it and can make it work, even though the generated artifacts do not end up in the correct place: I used to specify `--artifact-dir` but this cannot work due to the special mount paths (cf 
[ansible-runner-with-workaround.txt](https://github.com/ansible/ansible-runner/files/7802528/ansible-runner-with-workaround.txt)

@Shrews: thank you as well for your assistance. If patches are welcome, I could see to try and enable the option.

I am now stumbling upon a different problem: the inventory file. I see the playbook cannot parse the inventory file as specified:
```
[WARNING]: Unable to parse /runner/inventory/hosts as an inventory source
```

The problem is that, despite me specifying an argument to the `--inventory` option, it looks like the path is always overriden to be `/runner/inventory/hosts`, cf https://github.com/ansible/ansible-runner/blob/2.1.1/ansible_runner/config/runner.py#L169

Is this expected? I thought that the `--inventory` option served that purpose (to speficy the actual inventory path)?

**Comment by Shrews** [comment]
@vrubiolo Patches are _always_ welcome. We do require tests to validate any changes, though. That might be the most time consuming part.

**Comment by vrubiolo** [comment]
> @vrubiolo Patches are _always_ welcome. We do require tests to validate any changes, though. That might be the most time consuming part.

Thank you. Understood. This is indeed the right way of adding a feature and making it so that it can be maintained.

**Comment by Akasurde** [comment]
@vrubiolo Are you working on the fix?

**Comment by vrubiolo** [comment]
@Akasurde : sorry for replying so late, I got back to this ticket cleaning up for my AWX tasks.
Unfortunately, I am not working on a fix as of now, this will have to wait on my end.

**Comment by marshallford** [comment]
This bug seems to still to persist in [ansible-runner v2.3.4](https://github.com/ansible/ansible-runner/blob/2.3.4/ansible_runner/config/runner.py#L164). I don't understand why the inventory path is hardcoded when run with process isolation.

**Comment by vrubiolo** [comment]
Hi there and thanks for confirming the bug is still there @marshallford . On my side, I have moved off working on AWX so my need to getting this fixed is gone too.
